### PR TITLE
[doc] polish checkpoint docstring

### PR DESCRIPTION
**Pull request by ver217**

### CODE REVIEW

**Review by binmakeswell — APPROVED** [review]

File: colossalai/utils/checkpointing.py [hunk 1]
```diff
@@ -124,35 +125,27 @@ def partition_pipeline_parallel_state_dict(model, state_dict):
...
-    model: torch.nn.Module,
-    optimizer: torch.optim.Optimizer = None,
-    lr_scheduler: torch.optim.lr_scheduler._LRScheduler = None,
-    **kwargs
-):
+def save_checkpoint(file,
+                    epoch: int,
+                    model: torch.nn.Module,
+                    optimizer: torch.optim.Optimizer = None,
+                    lr_scheduler: torch.optim.lr_scheduler._LRScheduler = None,
```
Comment (feifeibear) [inline colossalai/utils/checkpointing.py:147]: Optional

File: colossalai/utils/checkpointing.py [hunk 2]
```diff
@@ -124,35 +125,27 @@ def partition_pipeline_parallel_state_dict(model, state_dict):
...
-    epoch: int,
-    model: torch.nn.Module,
-    optimizer: torch.optim.Optimizer = None,
-    lr_scheduler: torch.optim.lr_scheduler._LRScheduler = None,
-    **kwargs
-):
+def save_checkpoint(file,
+                    epoch: int,
+                    model: torch.nn.Module,
+                    optimizer: torch.optim.Optimizer = None,
```
Comment (feifeibear) [inline colossalai/utils/checkpointing.py:146]: Optional

File: colossalai/utils/checkpointing.py (hunk 1 above)
Comment (ver217) [reply colossalai/utils/checkpointing.py:147]: @kurisusnowdeng I notice that you comment codes abourt save/load optimizer and lr scheduler. Why?
